src/store/selectors/activities/index.ts

```diff
@@ -0,0 +1,439 @@
+import { Big } from "big.js";
+
+import {
+  Operation,
+  OperationType,
+  MintType,
+  MintOperation
+} from "@raha/api-shared/dist/models/Operation";
+
+import {
+  Activity,
+  VideoReference,
+  ActivityDirection,
+  ActivityContent,
+  PredeterminedBodyType
+} from "./types";
+import { getMemberById } from "../members";
+import { RahaState } from "../../reducers";
+import {
+  Member,
+  RAHA_BASIC_INCOME_MEMBER,
+  GENESIS_MEMBER,
+  GENESIS_TRUST_OPS,
+  GENESIS_REQUEST_INVITE_OPS
+} from "../../reducers/members";
+import {
+  CurrencyValue,
+  CurrencyRole,
+  CurrencyType
+} from "../../../components/shared/Currency";
+import { isUnconfirmedRequestInvite } from "../operations";
+import { List } from "immutable";
+import { MemberId } from "@raha/api-shared/dist/models/identifiers";
+
+function videoReferenceForMember(member: Member): VideoReference {
+  return {
+    videoUri: member.videoUri,
+    thumbnailUri: `${member.videoUri}.thumb.jpg`
+  };
+}
+
+/**
+ * Get all activities in Raha history.
+ * TODO: only get a subset, or at least a paginated list, of activities; this is
+ * not scalable.
+ *
+ * @returns Activities to render in reverse chronological order (new -> old)
+ */
+export function allActivities(state: RahaState): Activity[] {
+  return convertOperationsToActivities(state, state.operations).reverse();
+}
+
+/**
+ * Interpret a list of operations as a list of activities.
+ * TODO: make this more sophisticated, so that it's not just a one-to-one
+ * mapping from ops to activities
+ *
+ * @param operations All operations, expected in chronological order, oldest
+ * first.
+ */
+function convertOperationsToActivities(
+  state: RahaState,
+  operations: List<Operation>
+): Activity[] {
+  return operations.reduce(
+    (memo, operation) => {
+      const creatorMember = GENESIS_TRUST_OPS.includes(operation.id)
+        ? GENESIS_MEMBER
+        : getMemberById(state, operation.creator_uid);
+      if (!creatorMember) {
+        console.error(
+          `Operation with missing creator (id: ${
+            operation.creator_uid
+          }), invalid.`,
+          JSON.stringify(operation)
+        );
+        return memo;
+      }
+
+      switch (operation.op_code) {
+        case OperationType.CREATE_MEMBER: {
+          const newActivity: Activity = {
+            id: operation.id,
```

Comment:
They just need to be unique when they're rendered since they're used as keys for React, so maybe we can just make them a bunch of operation ids concatted. alternatively, I'll remove hte ID field, keep a list of participant operation ids along wiht the activity, and derive the key from that—probably better